lib/neovim-plugin.nix

```diff
@@ -157,7 +166,55 @@
                 ])
                 ++ (lib.optionals hasConfigAttrs [
                   (lib.optionalAttrs callSetup { ${namespace}.${name}.luaConfig.content = setupCode; })
-                  (lib.optionalAttrs (configLocation != null) (setLuaConfig cfg.luaConfig.content))
+                  (lib.mkIf (!cfg.lazyLoad.enable) (
+                    lib.optionalAttrs (configLocation != null) (setLuaConfig cfg.luaConfig.content)
+                  ))
+                ])
+                ++ (lib.optionals hasConfigAttrs [
+                  (lib.mkIf (config.plugins.lz-n.enable && cfg.lazyLoad.enable && cfg.lazyLoad.backendSettings != { })
```

Comment:
Maybe this should be nested conditions?

A top-level `mkIf cfg.lazyLoad.enable` that then has assertions and conditionals within the definition?
```nix
(mkIf cfg.lazyLoad.enable {
  plugins.lz-n = mkIf config.plugins.lz-n.enable {};
  plugins.lazy = mkIf config.plugins.lazy.enable {};
  assertions = [
    { assertion = cfg.lazyLoad.backendSettings != { }; }
    { assertion = length (filter (x: config.plugins.${x}.enable) [ "lz-n" "lazy" ]) == 1; }
  ];
})
```